### PR TITLE
feat: improve model reconcile telemetry

### PR DESCRIPTION
**Pull request by k-anshul**
closes https://linear.app/rilldata/issue/PLAT-389/improve-model-reconcile-telemetry

**Note**: Metrics in case of reconcile failures are already emitted for all resources so no metrics were added in the model reconciler for failures. 

**Checklist:**
- [x] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [x] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [x] I'm proud of this work!

### CODE REVIEW

File: runtime/reconcilers/model.go [hunk 1]
```diff
@@ -350,6 +355,55 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, n *runtimev1.ResourceNa
...
+				bytesAdded = newBytesTotal
+			}
+
+			// Persist updated totals
+			model.State.RowsTotal = newRowsTotal
+			model.State.BytesTotal = newBytesTotal
+			err = r.C.UpdateState(ctx, self.Meta.Name, self)
+			if err != nil {
+				return runtime.ReconcileResult{Err: err}
+			}
```
Comment (begelundmuller) [inline runtime/reconcilers/model.go]: One way or another, I think this should be part of `updateStateAfterExecution` – since the point of that function is to make the write atomic and to avoid many small DB writes

File: runtime/reconcilers/model.go [hunk 2]
```diff
@@ -350,6 +355,55 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, n *runtimev1.ResourceNa
...
+			var rowsAdded, bytesAdded int64
+			if firstRunIncremental {
+				rowsAdded = newRowsTotal - oldRowsTotal
+				if rowsAdded < 0 {
+					rowsAdded = 0
+				}
+				bytesAdded = newBytesTotal - oldBytesTotal
+				if bytesAdded < 0 {
+					bytesAdded = 0
+				}
```
Comment (begelundmuller) [inline runtime/reconcilers/model.go]:
```suggestion
				rowsAdded = max(newRowsTotal - oldRowsTotal, 0)
				bytesAdded = max(newBytesTotal - oldBytesTotal, 0)
```

File: runtime/reconcilers/model.go [hunk 3]
```diff
@@ -319,8 +320,12 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, n *runtimev1.ResourceNa
 		}
 	}
 
+	// Capture old row/byte totals before execution (for computing deltas)
+	oldRowsTotal := model.State.RowsTotal
+	oldBytesTotal := model.State.BytesTotal
```
Comment (begelundmuller) [inline runtime/reconcilers/model.go:325]: nit: I think this can be moved into the nested scope around line 354

File: runtime/reconcilers/model.go [hunk 4]
```diff
@@ -668,6 +708,12 @@ func (r *ModelReconciler) updateStateAfterExecution(
 	} else {
 		model.State.TotalExecutionDurationMs = model.State.LatestExecutionDurationMs
 	}
+	if rowsTotal > 0 {
+		model.State.RowsTotal = rowsTotal
+	}
+	if bytesTotal > 0 {
+		model.State.BytesTotal = bytesTotal
+	}
```
Comment (begelundmuller) [inline runtime/reconcilers/model.go:716]: Why not this?

```suggestion
	model.State.RowsTotal = rowsTotal
	model.State.BytesTotal = bytesTotal
```

Comment (k-anshul) [reply runtime/reconcilers/model.go:716]: This is required. If `queryOutputTableStats` fails we don't want to update older vals (though it should ideally not fail unless ctx errors).

File: runtime/reconcilers/model.go (hunk 3 above)
Comment (k-anshul) [reply runtime/reconcilers/model.go:325]: The first partitions updates the state. Ideally we want to capture rows before that.
